### ISSUE TITLE
Few Time Expressions yields errors [wip]

### ISSUE THREAD

**Issue by jas-valgotar** [microsoft/Power-Fx#602]
Below expression with Time function gives incorrect result.
Expression: "Time(2, 1, 1) + 60000" OR "60000 + Time(2, 1, 1)"
Current O/P: Error
Expected O/P: DateTime(2064,4,8,2,1,1)
Explanation: the addition should add 60000 days to 1970-01-01T10:02:01.000Z, which should result in  DateTime(2064,4,8,2,1,1)

// Returns milliseconds
>> Time(2, 1, 1) + -Time(2, 0, 1)
Current O/P: Error
Expected O/P: Time(0,1,0)

**Comment by LucGenetier** [comment]
Date reference isn't 1st Jan 1970 but approx. 1st Jan 1900
See https://docs.microsoft.com/en-us/office/troubleshoot/excel/wrongly-assumes-1900-is-leap-year
(We have that bug and need to keep it)

Adding 60k days to 1/1/70 would end up in 2134 (April11th)

**Comment by MikeStall** [comment]
From McCall, also consider: 

Time - Time = Time + (-Time)
And
Date - Date = Date + (-Date)

This may have some relation to #354 .

**Comment by MikeStall** [comment]
Is this addressed by Carlos's recent work?

**Comment by jas-valgotar** [comment]
Yes, this was fixed via, https://github.com/microsoft/Power-Fx/pull/659
